qcengine/cli.py

```diff
@@ -4,29 +4,147 @@
 
 import argparse
 import json
+import os.path
+import sys
 
-from . import compute
+from . import compute, compute_procedure  # run and run-procedure
+from qcelemental.models import ResultInput # run and run-procedure
+from . import __version__, list_all_procedures, list_all_programs, list_available_procedures, \
+    list_available_programs, get_procedure, get_program  # info
+from .config import global_repr  # info
 
 __all__ = ["main"]
 
 
 def parse_args():
     parser = argparse.ArgumentParser(description='A CLI for the QCEngine.')
+    parser.add_argument('--version', action='version', version=f"{__version__}")
 
-    parser.add_argument("program", type=str, default=None, help="The program to call")
-    parser.add_argument("data", type=str, default=None, help="A JSON blob of the task description.")
+    subparsers = parser.add_subparsers(dest="command")
 
-    return vars(parser.parse_args())
+    info = subparsers.add_parser('info', help="Print information about QCEngine setup, version, and environment.")
+    info.add_argument('--version', action='store_true', help="Print version of QCEngine and QCElemental.")
+    info.add_argument('--programs', action='store_true', help="Print detected and supported programs.")
+    info.add_argument('--procedures', action='store_true', help="Print detected and supported procedures.")
+    info.add_argument('--config', action='store_true', help="Print host, compute, and job configuration.")
+    info.add_argument('--all', action='store_true', help="Print all available information.")
```

Comment:
To make this simpler I like the idea of a:
`QCEngine info config`

Can be implemented as:

```
info.add_argument("category", nargs="?", default="all", choices=["version", "programs", ...], help="The config category to show.")
```

Reply:
DOne.